mono/utils/mono-logger-internals.h

```diff
@@ -17,6 +17,7 @@ typedef enum {
 	MONO_TRACE_THREADPOOL		= (1<<7),
 	MONO_TRACE_IO_THREADPOOL	= (1<<8),
 	MONO_TRACE_IO_LAYER		= (1<<9),
+	MONO_TRACE_W32HANDLE	= (1<<10),
```

Comment:
We want seperate logging because the io-layer is comprised of at least 2 things: win32 api implementation and win32 handle lifetime + waiting + signalling.
By having everything together, it makes it harder to select exactly what you want to log, as for example you might try to track down a Event handle leak (which has nothing to do with opening and closing files), but the log will be cluttered with all this noise.
So I would strongly lean toward splitting it, but if that means stopping this PR to be merged, then I will revert it.